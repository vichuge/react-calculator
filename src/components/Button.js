import React from 'react';
import PropTypes from 'prop-types';

const Button = (props) => {
  const { buttonName, type } = props;

  return (
    <button type="button" className={type}>{buttonName}</button>
  );
};

Button.propTypes = {
  buttonName: PropTypes.string,
  type: PropTypes.string,
};

Button.defaultProps = {
  buttonName: 'bName',
  type: 'dType',
};

export default Button;
