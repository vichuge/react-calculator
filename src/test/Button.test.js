import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../components/Button';

it('return button panel', () => {
  const clickHandle = (buttonName) => {
    console.log(buttonaName);
  };
  const tree = renderer
    .create(<Button buttonName="AC" clickPanel={clickHandle} color="gray" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});