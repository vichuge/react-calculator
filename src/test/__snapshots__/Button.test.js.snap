// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`return button panel 1`] = `
<button
  className="gray "
  onClick={[Function]}
  type="button"
>
  AC
</button>
`;
